sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/AADClientRegistrationRepository.java

```diff
@@ -3,83 +3,402 @@
+        return toClientRegistrationBuilder(registrationId, clientProperties).build();
+    }
+
+    private boolean isAzureDelegatedClientRegistration(String registrationId,
+                                                       AuthorizationClientProperties clientProperties) {
+        return !AZURE_CLIENT_REGISTRATION_ID.equals(registrationId)
+            && AUTHORIZATION_CODE.equals(clientProperties.getAuthorizationGrantType())
+            && !clientProperties.isOnDemand();
     }
 
     @Override
     public ClientRegistration findByRegistrationId(String registrationId) {
-        return allClients.get(registrationId);
+        Assert.hasText(registrationId, "registrationId cannot be empty");
+        return allClientRegistrations.get(registrationId);
     }
 
-    public AzureClientRegistration getAzureClient() {
-        return azureClient;
+    @NotNull
+    @Override
+    public Iterator<ClientRegistration> iterator() {
+        return Collections.singleton(azureClientRegistration.getClient()).iterator();
     }
 
-    public boolean isClientNeedConsentWhenLogin(ClientRegistration client) {
-        return otherClients.contains(client)
-            && properties.getAuthorizationClients().get(client.getClientName()) != null
-            && !properties.getAuthorizationClients().get(client.getClientName()).isOnDemand();
+    public AzureClientRegistration getAzureClientRegistration() {
+        return azureClientRegistration;
     }
 
-    public boolean isClientNeedConsentWhenLogin(String id) {
-        ClientRegistration client = findByRegistrationId(id);
-        return client != null && isClientNeedConsentWhenLogin(client);
+    public boolean isAzureDelegatedClientRegistrations(ClientRegistration client) {
+        return delegatedClientRegistrations.containsValue(client);
     }
 
-    public static boolean isDefaultClient(ClientRegistration clientRegistration) {
-        return AZURE_CLIENT_REGISTRATION_ID.equals(
-            clientRegistration.getClientName());
+    public boolean isAzureDelegatedClientRegistrations(String registrationId) {
+        return delegatedClientRegistrations.containsKey(registrationId);
     }
 
     public boolean isClientCredentials(String registrationId) {
-        return Optional.ofNullable(allClients.get(registrationId))
+        return Optional.ofNullable(allClientRegistrations.get(registrationId))
              .map(ClientRegistration::getAuthorizationGrantType)
              .map(AuthorizationGrantType::getValue)
              .map(v -> v.equals(AADAuthorizationGrantType.CLIENT_CREDENTIALS.getValue()))
              .orElse(false);
     }
 
-    public static boolean isDefaultClient(String clientId) {
-        return AZURE_CLIENT_REGISTRATION_ID.equals(clientId);
+    public static boolean isDefaultClient(String registrationId) {
+        return AZURE_CLIENT_REGISTRATION_ID.equals(registrationId);
+    }
+
+
+    public boolean isWebAppClientMode() {
```

Comment:
You already created `AADConditions` why not reuse it here?

Reply:
You are right, but I need to study how to reuse the SpringBootCondition after the bean register.

Reply:
After I checked, it can not be accessed for the parameter class `ConditionContext ` of the method `org.springframework.context.annotation.Condition#matches(ConditionContext, AnnotatedTypeMetadata)`,  I just think this feature is only available in the spring bean initializing.

Reply:
isWebApplicationOnly

Reply:
OK